pkg/kube/container_files.go

```diff
@@ -76,23 +74,24 @@ func (c *Container) addLogDirectory() error {
 }
 
 func (c *Container) addOCIBundle() error {
-	glog.V(8).Infof("Creating bundle directory %s", c.bundlePath())
-	err := os.MkdirAll(c.bundlePath(), 0755)
+	glog.V(8).Infof("Creating SIF bundle at %s", c.bundlePath())
+	d, err := ocibundle.FromSif(c.imgInfo.Path(), c.bundlePath(), true)
 	if err != nil {
-		return fmt.Errorf("could not create bundle directory for container: %v", err)
+		return fmt.Errorf("could not create SIF bundle driver: %v", err)
 	}
-	if err := c.prepareOverlay(c.imgInfo.Path()); err != nil {
-		return err
+	if err := d.Create(nil); err != nil {
```

Comment:
you need to pass the OCI configuration instead of nil

Reply:
If I pass oci config then users are not set correctly with respect to container's `/etc/passwd`

Reply:
More specifically, this will not be evaluated as expected: https://github.com/sylabs/singularity-cri/blob/master/pkg/kube/container_oci.go#L337

Reply:
Yes I mean you can call `d.Create(ociSpec)` at location where `os.OpenFile` is, same effect and you can get rid of writing file since ocibundle do it for you

Reply:
but spec should be generated when image is mounted, so we can read container's `/etc/passwd` and `/etc/group` to set user in OCI config

Reply:
Oh yeah right !